toolchain/check/testdata/interop/cpp/function/int16.carbon

```diff
@@ -6,115 +6,115 @@
 //
 // AUTOUPDATE
 // TIP: To test this file alone, run:
-// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/interop/cpp/function/param_int16.carbon
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/interop/cpp/function/int16.carbon
```

Comment:
From discussion about thunking, we expect a lot of types to behave in a similar way:

- i32 and i64 should work without a thunk
- i8, i16, and i128 should require a thunk
- Similar for unsigned

 I'm hesitant to get into a pattern where each supported type has its own file (e.g., int8.carbon, int64.carbon, int128.carbon, double.carbon, even the float.carbon I just mentioned; is there also char.carbon, etc?). If your renaming, perhaps filenames should be closer to categories of behavior, rather than individual types?

Reply:
"Arithmetic" alone would mean `+-*/`; how about `numbers` or `arithmetic_types`? cppreference says [arithmetic types](https://en.cppreference.com/w/c/language/arithmetic_types.html), which gives weight to that option.

Also, why is `float` still separate in return.carbon?